### DISCUSSION TITLE
Where to put favicon.ico in a simple app?

### DISCUSSION THREAD

**Discussion by Kugeleis** [widgetti/solara#710] [Q&A]
I serve solara from just the top level app.py. Where should I put the favicon.ico to be found by solara. I tried the same level as app.py and in a subfolder `assets/favicon.ico` and both didn't work.

**Comment by maartenbreddels** [comment]
Hi,

See https://solara.dev/documentation/advanced/reference/asset-files
Does that answer your question?

**Comment by Kugeleis** [comment]
That's where I looked. Does this mean I always need a `pages` subfolder and the favicon is always at ./../assets/ relative to it?

**Reply by maartenbreddels** [reply]
Yes, although the pages directory can be anything you'd like.

**Reply by Kugeleis** [reply]
I tried
```
assets_path = os.path.abspath(os.path.dirname(__file__)) + "/assets"
settings.assets.extra_locations.append(assets_path)
```
but it didn't see the favicon.ico there ;(

**Reply by maartenbreddels** [reply]
What do you mean by `but it didn't see the favicon.ico there ;(`
We will not put the icon file there, you have to put it in the assets folder.

**Reply by Kugeleis** [reply]
I have
```
- app.py
|_ assets
  - favicon.ico
  - ...
```
and I thought with `solara.server.settings.assets.extra_locations.append(assets_path)` I could tell solara to look for assets there.

**Reply by maartenbreddels** [reply]
Yeah, that should work!
Which version of solara are you running?

**Reply by maartenbreddels** [reply]
Btw, we don't use favicon.ico but favicon.png (and also favicon.svg, which I see is missing from that documentation page)

Can you try making a png icon?

**Reply by Kugeleis** [reply]
I just updated to version 1.35.1. No favicon.png is not loaded despite `settings.assets.extra_locations == ['/home/kugeleis/solara-page/assets', 'assets', './assets']`.

**Reply by maartenbreddels** [reply]
With this setup:
```
├── assets
│   ├── favicon.png
│   └── favicon.svg
└── test.py
```

```python
import os
import solara
import solara.server.settings

assets_path = os.path.abspath(os.path.dirname(__file__)) + "/assets"
solara.server.settings.assets.extra_locations.insert(0, assets_path)

clicks = solara.reactive(0)


@solara.component
def Page():
    color = "green"
    if clicks.value >= 5:
        color = "red"

    def increment():
        clicks.value += 1
        print("clicks", clicks)  # noqa

    solara.Button(label=f"Clicked: {clicks}", on_click=increment, color=color)
```

I managed to get it working.
However, the browser likes to cache things, so I also saw the previous favicon before. To avoid that, you can try running on a different port (solara run --port 8777) or open the dev console and make sure Network->Disable cache is checked, and refresh.

Hope you also get it working.

**Reply by Kugeleis** [reply]
This only works if I have a SVG favicon there. If this is missing the standard solara symbol wins despite my PNG there. I'd consider this a bug.

**Reply by maartenbreddels** [reply]
why do you consider this a bug? It's what browser do when they encounter:
```
    <link rel="icon" type="image/png" sizes="any" href="{{root_path}}/static/assets/favicon.png">
    <link rel="icon" type="image/svg+xml" href="{{root_path}}/static/assets/favicon.svg">
```

It's up to the browser to choose. We should update the docs on this, but I don't see this as a bug? Do you agree?

**Reply by Kugeleis** [reply]
Well, if svg always wins what is the point having an png at all?

**Reply by maartenbreddels** [reply]
As I said, it can depend on the browser. Maybe you can check the current browser behavior and tell us if it does or does not make sense to have both. There is quite some StackOverflow content to go through, would appreciate some help here.

**Reply by Kugeleis** [reply]
Having tested Chromium (Edge and Chrome) and Firefox on Windows - all ignore the png.

**Reply by hkayabilisim** [reply]
I've come across exactly the same situation. I had only a favicon.png not SVG. So, when I put it the favicon.png under assets folder, I couldn't change the favicon because all browsers (Opera, Safari and Firefox) used the favicon.svg provided in the solara server. So, I had to create my favicon.svg and put it under assets folder, and the problem is solved.

So, in the light of my experience, I believe it is not a bug, but in the documentation it could be explained that the server provides a favicon.svg which overrides PNG file.  

Created a PR for the improved doc: https://github.com/widgetti/solara/pull/796
